front/pages/w/[wId]/builder/assistants/new.tsx

```diff
@@ -256,20 +263,52 @@ export default function CreateAssistant({
       throw new Error("Form not valid");
     }
 
-    let tfParam: t.TypeOf<
+    type BodyType = t.TypeOf<
       typeof PostOrPatchAgentConfigurationRequestBodySchema
-    >["assistant"]["action"]["timeframe"] = "auto";
-
-    if (timeFrameMode === "FORCED") {
-      if (!timeFrame.value) {
-        // unreachable
-        // we keep this for TS
-        throw new Error("Form not valid");
-      }
-      tfParam = {
-        duration: timeFrame.value,
-        unit: timeFrame.unit,
-      };
+    >;
+
+    let actionParam: BodyType["assistant"]["action"] | null = null;
+    switch (dataSourceMode) {
+      case "GENERIC":
+        break;
+      case "SELECTED":
+        let tfParam: NonNullable<BodyType["assistant"]["action"]>["timeframe"] =
```

Comment:
`NonNullable` here allows me to extract the type of "timeframe" even though ["assistant"]["action"] can be null